### PR TITLE
Updated user crud for redirecting accordingly to the user action.

### PR DESCRIPTION
**Pull request by vinyll**

### CODE REVIEW

File: static/js/utils.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+  function request (url, params) {
```
Comment (davidbgk) [inline static/js/utils.js]: Let's steal it from https://framagit.org/drone/gallinipper/blob/master/js/request.js ;)

Comment (vinyll) [reply static/js/utils.js]: I'd keep it for when we need further functionalities on our lib. We have 20 lines that are pretty lite right now doing just what we need. Open to it if we're missing something here though. 🌮

Comment (davidbgk) [reply static/js/utils.js]: That version is not dealing with server being inaccessible

File: templates/user.html [hunk 2]
```diff
@@ -1,64 +1,76 @@
...
+    document.querySelector('.user-save').addEventListener('submit', event => {
+      event.preventDefault()
 
-  const fields = Array.from(document.querySelectorAll('[data-edit]'))
-  const data = {
-    'firstname': fields.find(f => f.dataset.edit === 'firstname').textContent,
-    'lastname': fields.find(f => f.dataset.edit === 'lastname').textContent,
-    'email': fields.find(f => f.dataset.edit === 'email').textContent,
-  }
+      const fields = Array.from(document.querySelectorAll('[data-edit]'))
```
Comment (davidbgk) [inline templates/user.html]: I wonder if we should convert it to something like [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData)

File: templates/user.html [hunk 3]
```diff
@@ -1,64 +1,76 @@
...
+        'email': fields.find(f => f.dataset.edit === 'email').textContent,
+      }
+
+      request(event.target.action, {
+        method: event.target.dataset.method || event.target.method,
+        body: data
+      })
+      .then(r => {
+        location.href = `/user/${r.body.id}/`
+      })
```
Comment (davidbgk) [inline templates/user.html]: No `.catch()`?

Comment (vinyll) [reply templates/user.html]: Error catching are on their way 🏃

File: templates/user.html [hunk 4]
```diff
@@ -1,64 +1,76 @@
...
+    })
+
+    const deleteForm = document.querySelector('.user-delete')
+    deleteForm && deleteForm.addEventListener('submit', event => {
+      event.preventDefault()
+      request(event.target.action, {method: event.target.dataset.method})
+      .then(r => {
+        if(r.status === 204) {
+          console.info('user deleted')
+          location.href = '/'
```
Comment (davidbgk) [inline templates/user.html]: We might want to use flash messages server-side when we then redirect (see PR#1)

File: templates/user.html (hunk 2 above)
Comment (vinyll) [reply templates/user.html]: And then to _JSON_?

File: static/js/utils.js (hunk 1 above)
Comment (vinyll) [reply static/js/utils.js]: ok ok, I'll give it a shot

**Pull request merged**